README.md

```diff
@@ -21,47 +22,49 @@ we store those as well:
 
 ```bash
 buildtest/tools/buildsystem/schemas
-   singlesource
-   singlecompile
+   script
 ```
 
+The above would include a folder of schemas for the "script" type, each named by a version.
+For example:
+
+
+```bash
+script/
+    script-v0.0.1.schema.json
+    script-v0.0.2.schema.json
+    ...
+```
+
+
 ## How is this repository organized?
 
 The entire root of this folder could be added to the folder shown above,
 and the subfolder structure would be added to the module. Importantly,
 the lowercase class names should match to a class defined in in [buildtest/tools/buildsystem/bases.py]().
-For example, the contents of the [singlesource](singlesource) directory here are expected 
-to be added present at `buildtest/tools/buildsystem/schemas/singlesource` to go
-along with a class "SingleSource" in the bases file. This is
-the default for a buildsystem model, and it can be changed, however
-this is not recommended.
-
+For example, the contents of the [script](script) directory here are expected 
+to be added present at `buildtest/tools/buildsystem/schemas/script` to go
+along with a class "Script" in the bases file. 
 
 ## What are shared attributes?
 
 Every build configuration across types can share some high level metadata,
 including a type, description, and maintainer. Additionally, each attribute
 should have a key with a unique name so that more than one configuration
-can be represented in one file. Here is a quick example for a [singlecompile](singlecompile)
+can be represented in one file. Here is a quick example for a [script](script)
 example:
 
 
 ```yaml
-# module load example with one version
+version: 0.0.1
 hello_ex1:
-  type: singlesource
-  description: "C hello world example1"
-  compiler: gnu
-  source: hello.c
-  scheduler: local
-  moduleload:
-    key:
-      name: GCC
-      version: ["6.4.0"]
-  cflags:
-    - "-O1"
+  type: script
+  description: "hello world example"
+  maintainer: "@vsoch"
```

Comment:
Does it make sense to have `maintainer` inside each sub-test. I would think if someone is writing a YAML file, they are the maintainer of all sub-test. So i am proposing having `maintainer` outside so its defined once per file. 
I was thinking instead of ``maintainer: "@vsoch"``  --> ``maintainer: "First Last <vsochat@stanford.edu>"``. The schema  can check the maintainer field with value of ``git config user.name`` and ``git config user.email``.

It's just a thought. I think it makes sense user can track test to their email address and a name than using a github username.

Reply:
hmm, to play devil's advocate, I could imagine someone copy pasting someone else's recipe into a common file of tests. I generally think metadata about the test should be placed with the test, and not outside of it with the file.

Reply:
And I disagree about trying to use git to automatically add a maintainer - it's the user's choice to add or not. Adding this information when they don't provide it feels like a violation of privacy. If they want to provide an alias, they will.